base/abstractarray.jl

```diff
@@ -343,6 +343,27 @@ function first(itr)
     x[1]
 end
 
+"""
+    first(v::AbstractVector, n::Integer)
+
+Get the first `n` elements of vector `v`, or fewer elements if `v` is not long enough.
+
+# Examples
+```jldoctest
+julia> first(["foo", "bar", "qux"], 2)
+2-element Array{String,1}:
+ "foo"
+ "bar"
+
+julia> first(1:6, 10)
+1:6
+
+julia> first(Bool[], 1)
+0-element Array{Bool,1}
+```
+"""
+first(v::AbstractVector, n::Integer) = @inbounds v[firstindex(v):min(firstindex(v) + n - 1, end)]
```

Comment:
You can use `begin` instead of `firstindex`

Reply:
Nice, now the two definitions are much more symmetric